### PR TITLE
Introduce black formatter

### PR DESCRIPTION
**Pull request by no2chem**
This PR introduces the black formatter as a pre-commit hook. 

When you run `git commit`, black automatically re-formats the code before passing it to flake8.

example:
```
vscode ➜ /workspaces/wideq (format ✗) $ git commit -a -m "introduce black formatter"
black....................................................................Passed
Flake8...................................................................Passed
```

There are some differences in how black and flake8 interpret pep. I'm not really opinionated here as I am not a python expert, but it seems that the black authors think they are right in interpreting the PEP, and black fixes the formatting in a consistent way that an explicit rule shouldn't be required. The exceptions are defined in the `.flake8` file.

This change should make it easier to compare changes, as small style differences should be make consistent by the tool. For example, #132 has a lot of changes because I ran `black` just to correct the `flake8` errors. This eliminates formatting differences from pull requests.

You can always skip the precommit hooks by running
```
git commit --no-verify
```

### CODE REVIEW

**Comment by no2chem** [comment]
well, it seems to really not want to run automatically on the test directory, not sure why...

**Comment by sampsyo** [comment]
Hi there—although I'm not a frequent Black user, I'm OK with adding it if users are excited about automatic formatting. (TBH, I find gentle feedback from flake8 enough to ensure consistency, although I know some folks really get a lot of value out of automatic reformatters.)

If we're going to do this, however, I think it would be best to enforce its use "server-side" rather than relying on folks to use the pre-commit hook. That would mean reconfiguring [our GitHub actions](https://github.com/sampsyo/wideq/blob/master/.github/workflows/style.yml) to run something like `black --check` or `black-primer` in addition to flake8 and mypy. Would anybody be interested in taking care of that?

File: pyproject.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["flit"]
+requires = ["flit", "pre-commit"]
```
Comment (sampsyo) [inline pyproject.toml:2]: I don't think this should be considered a build-time dependency.

File: README.md [hunk 2]
```diff
@@ -30,6 +30,11 @@ You can also specify one of several other commands:
 * `turn <ID> <ONOFF>`: Turn an AC device on or off. Use "on" or "off" as the second argument.
 * `ac-config <ID>`: Print out some configuration information about an AC device.
 
+Development
+-------
+To ensure consistent formatting across pull requests, install the precommit hooks to auto format your code using `pre-commit install`.
+
+The code will be auto-formatted by `black` to ensure consistent style.
```
Comment (sampsyo) [inline README.md:37]: I'm gong to rephrase this a bit to discuss the use of Black more broadly.

**Pull request merged**